kmmsdk/src/commonMain/kotlin/com/lokate/kmmsdk/LokateSDK.kt

```diff
@@ -1,103 +1,227 @@
+        lokateScope.launch {
+            while (true) {
+                delay(GONE_CHECK_INTERVAL)
+                log.d { "Checking for gone beacons" }
                 activeBeacons.forEach {
-                    if (it.lastSeen < getTimeMillis() - DEFAULT_TIMEOUT_BEFORE_GONE) {
-                        emit(it)
+                    if (it.seen < getTimeMillis() - Defaults.DEFAULT_TIMEOUT_BEFORE_GONE) {
+                        gone.emit(it)
+                        activeBeacons.remove(it)
                     }
                 }
             }
+        }
+    }
 
+    private fun flowEmitter() {
         lokateScope.launch {
-            beaconFlowHandler(newComer, alreadyIn, gone)
+            newComer.collect {
+                log.d { "Newcomer: $it" }
+                beaconRepository.sendBeaconEvent(
+                    EventRequest(
+                        "umut",
+                        it.beaconUUID,
+                        it.major.toString(),
+                        it.minor.toString(),
+                        EventStatus.ENTER,
+                        it.seen,
+                    ),
+                ).also { log.d { "Send newcomer in: $it" } }
+            }
+        }
+        lokateScope.launch {
+            alreadyIn.collect {
+                log.d { "Already in: $it" }
+                beaconRepository.sendBeaconEvent(
+                    EventRequest(
+                        "umut",
+                        it.beaconUUID,
+                        it.major.toString(),
+                        it.minor.toString(),
+                        EventStatus.STAY,
+                        it.seen,
+                    ),
+                ).also { log.d { "Send already in: $it" } }
+            }
+        }
+        lokateScope.launch {
+            gone.collect {
+                log.d { "Gone: $it" }
+                beaconRepository.sendBeaconEvent(
+                    EventRequest(
+                        "umut",
+                        it.beaconUUID,
+                        it.major.toString(),
+                        it.minor.toString(),
+                        EventStatus.EXIT,
+                        it.seen,
+                    ),
+                ).also { log.d { "Send gone: $it" } }
+            }
         }
-        // beaconScanner.start()
-        // results = beaconScanner.observeResults()
     }
 
-    private suspend fun beaconFlowHandler(
-        beaconScannerFlow: Flow<BeaconScanResult>,
-        alreadyIn: Flow<BeaconScanResult?>,
-        gone: Flow<BeaconScanResult>,
-    ) {
-        // first look for the beacons that are already in the list
-        alreadyIn.collect {
-            it?.let {
-                activeBeacons.removeAll {
+    private fun scanResultHandler(beaconScannerFlow: Flow<BeaconScanResult>) {
+        lokateScope.launch {
+            beaconScannerFlow.collect { scan ->
+                activeBeacons.firstOrNull {
                     it.beaconUUID == it.beaconUUID &&
```

Comment:
it should be: it.beaconUUID == scan.beaconUUID. below two lines also should be changed accordingly

Reply:
Yes thats a bug will fix that